ProcessLib/HT/HTProcess.cpp

```diff
@@ -160,5 +165,54 @@ void HTProcess::setCoupledTermForTheStaggeredSchemeToLocalAssemblers()
         _local_assemblers, _coupled_solutions);
 }
 
+NumLib::LocalToGlobalIndexMap* HTProcess::getDOFTableForExtrapolatorData(
+    bool& manage_storage) const
+{
+    if (!_use_monolithic_scheme)
+    {
+        // For single-variable-single-component processes reuse the existing DOF
+        // table.
+        manage_storage = false;
+        return _local_to_global_index_map.get();
+    }
+
+    // Otherwise construct a new DOF table.
+    std::vector<MeshLib::MeshSubsets> all_mesh_subsets_single_component;
+    all_mesh_subsets_single_component.emplace_back(
+        _mesh_subset_all_nodes.get());
+
+    manage_storage = true;
+
+    return new NumLib::LocalToGlobalIndexMap(
+        std::move(all_mesh_subsets_single_component),
+        // by location order is needed for output
+        NumLib::ComponentOrder::BY_LOCATION);
+}
+
+void HTProcess::setCoupledSolutionsOfPreviousTimeStep()
+{
+    const auto number_of_coupled_solutions =
+        _coupled_solutions->coupled_xs.size();
+    _coupled_solutions->coupled_xs_t0.clear();
+    _coupled_solutions->coupled_xs_t0.reserve(number_of_coupled_solutions);
+    const int process_id = _coupled_solutions->process_id;
+    for (std::size_t i = 0; i < number_of_coupled_solutions; i++)
+    {
+        const auto x_t0 = _xs_previous_timestep[process_id].get();
+        if (x_t0 == nullptr)
```

Comment:
Minor:
if the type of `_xs_previous_timestep[process_id]` is `std::unique_ptr` or similar, there is no need to `get` the pointer for comparison or later access:
```c++
auto const& x_t0 = xs[id];
if (!x_t0) OGS_FATAL(...);
```

Reply:
removed `.get()`.